tests/transforms/test_qcut.py

```diff
@@ -2323,24 +2325,156 @@ def test_circuit_with_disconnected_components(self, use_opt_einsum, mocker):
         dev = qml.device("default.qubit", wires=3)
 
+        @qml.transforms.cut_circuit(use_opt_einsum=use_opt_einsum)
+        @qml.qnode(dev)
+        def circuit(x):
+            qml.RX(x, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.WireCut(wires=1)
+            qml.CNOT(wires=[1, 2])
+            qml.RY(x**2, wires=2)
+            return qml.expval(qml.PauliZ(wires=[0]))
+
+        x = 0.4
+        res = circuit(x)
+        assert np.allclose(res, np.cos(x))
+
+    def test_circuit_with_trivial_wire_cut(self, use_opt_einsum, mocker):
+        """Tests that a circuit with a trivial wire cut (not separating the circuit into
+        fragments) is executed correctly"""
+        dev = qml.device("default.qubit", wires=2)
+
         @qml.transforms.cut_circuit(use_opt_einsum=use_opt_einsum)
         @qml.qnode(dev)
         def circuit(x):
             qml.RX(x, wires=0)
             qml.CNOT(wires=[0, 1])
             qml.WireCut(wires=0)
             qml.CNOT(wires=[0, 1])
-            qml.RY(x**2, wires=2)
             return qml.expval(qml.PauliZ(wires=[0]))
 
         spy = mocker.spy(qcut, "contract_tensors")
 
         x = 0.4
         res = circuit(x)
         assert np.allclose(res, np.cos(x))
-        assert len(spy.call_args[0][0]) == 1  # there should be 2 tensors for wire 0
+        assert len(spy.call_args[0][0]) == 1  # there should be 1 tensor for wire 0
         assert spy.call_args[0][0][0].shape == ()
 
+    def test_complicated_circuit(self, mocker, use_opt_einsum):
+        """
+        Tests that the full circuit cutting pipeline returns the correct value and
+        gradient for a complex circuit with multiple wire cut scenarios. The circuit is cut into
+        fragments of at most 2 qubits and is drawn below:
+
+        0: ──BasisState(M0)──//───────╭C───//──RX(0.40)─╭C───//─╭C────────╭//──────────────────
+        1: ───────────────────────────╰X────────────────╰X───//─╰Z────────╰//──────────────────
+        2: ──H──────────────╭C──────────────────────────────────╭RY(0.64)──────────────────────
+        3: ─────────────────╰RY(0.50)──//──H──╭C─────────//──H──╰C─────────//─╭RY(0.89)──//──H─
+        4: ───────────────────────────────────╰RY(0.36)──H────────────────────╰C───────────────
+
+        ──────────────────────────┤
+        ────────────────╭RX(0.60)─┤ ╭<Z@Z@Z>
+        ╭RY(0.88)───────│─────────┤ ├<Z@Z@Z>
+        ╰C──────────────│─────────┤ ╰<Z@Z@Z>
+        ────────────────╰C────────┤
+        """
+        dev_original = qml.device("default.qubit", wires=5)
+
+        # We need a 4-qubit device to account for mid-circuit measurements
+        dev_cut = qml.device("default.qubit", wires=4)
+
+        def two_qubit_unitary(param, wires):
+            qml.Hadamard(wires=[wires[0]])
+            qml.CRY(param, wires=[wires[0], wires[1]])
+
+        def f(params):
+            qml.BasisState(np.array([1]), wires=[0])
```

Comment:
If we did
```python
@qml.cut_circuit
@qml.qnode(dev)
def f(params):
    ...
```
Then it would be hard to separate the uncut circuit from the cut circuit - i.e., `f` would necessarily be the cut circuit and we'd have to have another function definition
```python
@qml.qnode(dev)
def f_uncut(params):
    ...
```
This would also work, just takes a bit more space (especially since the contents of `f` is quite large).